File: Nim_Game.cpp
class Solution {
public:
    bool canWinNim(int n) {
     //T(C(N)=O(1)) and S(C(N)=O(1)) as it requires not addiitonal space with in unit Time        
        if (n%4==0)return false;//Printing False to the stone heap last
        else return true;}};//Printing False to the First stone heap 

File: Nim_Game.py
class Solution:
    def canWinNim(self, n: int) -> bool:
        #T(C(N)==O(1)) and S(C(N)=O(1)) as it requires non additional Space in Unit time 
        if n%4==0: return False#printing false to the UNoptimal stone's heeap
        else: return True#Pritning True to FIrst Stone heap
